### ISSUE TITLE
Firmware 092+091 "missing steps"

### ISSUE THREAD

**Issue by Kulitorum** [repetier/Repetier-Firmware#345]
I upped the power on my printrboard to 19v a long time ago, and have been running with PID on 0.83 for years. I have never been able to get PID to work, so I got into a habbit of waiting for the extruder to overshoot to 250c when I ask it to goto 220c. Today I got tired of that and found out about dead-time calibration, and thought I'd give it a try.

So I installed 0.92 to test it out, but it skips some movements. After just 2-400 moves, the X axis does not move for 1/10 of a second or so, resulting in layer offset. This repeats every minute or so.

My instinct was of cause that the stepper driver was overheating, so I took the power down 20% and slowed the printer down to 50%, just to test (and lowered the acceleration) This made no change. I then added active cooling to the printrboard - no change.

I installed 0.91, and got the same result. - layers still shifts every minute or so.

I installed 0.83 again, and the problem went away (but bad PID is back)

Any ideas?

**Comment by Kalus1** [comment]
I am having a similar issue using both a Ramps 1.4 and a Rumba board.
Random y-axis skipping which appears to be at layer change. I've tried 0.91 and 0.92 with the same problem. I swapped over to Marlin with the all of the same settings and the problem disappears. I have tried setting STEPPER_HIGH_DELAY to 1 which 'seemed' to lessen the skipping. The last print  I tried it got to layer 8, then skipped in the same direction, about 0.5mm on each of the following 3 layers before I killed the job. I'm going to try STEPPER_HIGH_DELAY 2 to see if that changes anything.

I was using a Sanguino board with the same drivers and stepper motors previously with no issues at all. I bought the Rumba first, got frustrated and purchased a Ramps to replace it, but had the same problem.

Kulitorum, did you have any success??

**Comment by Kalus1** [comment]
by the way, I have separate power supplies for the bed and hot end.
Changing STEPPER_HIGH_DELAY to 2, actually did the same thing, except the amount of skipped about doubled. It started skipping on layer 10 this time though.
I might add, there is no sound, normally associated with a stepper skipping due to low/high current settings. 
I'm dumbfounded. I think I'll try 0.83 and see what happens.

**Comment by repetier** [comment]
Y axis on rumba is a heat problem of the rumba board and nothing firmware related. Use E2 for y axis and your problems are gone even with the same stepper driver you now have on y axis.

**Comment by Kalus1** [comment]
Ok, I may try the Rumba board again later and will keep it in mind, thank you.
I'm going to try printing from SD card now to see if that changes anything.

**Comment by Kalus1** [comment]
Printing from SD does the same thing.
This is the print from Repetier firmware 0.91...
![20150129_091542 1](https://cloud.githubusercontent.com/assets/1237561/5959959/b9bbaf60-a797-11e4-8356-0aa384621080.jpg)
and an immediate follow up print with Marlin...
![20150129_091627 1](https://cloud.githubusercontent.com/assets/1237561/5959974/d862d9c0-a797-11e4-8b26-8b1c7d637e10.jpg)

**Comment by Kalus1** [comment]
What is weird is it is happening on the same layer each time, starting at layer 11. I just checked the 5 previous prints of the same object. I had a look at the Gcode and can't see any shift at all.

**Comment by repetier** [comment]
Looks like different firmware settings. Hard to see from this perspective but is there a y shift? 
That was wis RAMPS right?

Never forget the values in eeprom are used not the one in configuration.h. Maybe it is jerk or y acceleration, both can also shift if too high.

**Comment by repetier** [comment]
So it might be the z change and move to starting point there. Guess it has a difference that causes the problem too trigger. Try reducing z jerk. What acceleration are you using for xy? I normally use 1000 to get smooth moves even if printer can go faster.

**Comment by Kalus1** [comment]
Yes, this is all with the Ramps board. I had just been using the defaults, but have now just set all acceleration to 500mm/s2 for xy and 50mm/s2 for z, xy jerk to 10.0 and got a successful print. I will do some more prints of the same object when I get home from work to see if I can raise these and see what happens.
Thank you for your help, I feel relieved :)
What is the big difference between Marlin and Repetier with regards to acceleration and jerk?
I will post some results as I go, so maybe it will help others if they have the same problem.

**Comment by repetier** [comment]
Acceleration should be the same. 
Big question is how change from xy move to z move is handled. I trade them as 2 different problems that both should be satisfied.I guess Marlin does the same, but I'm not that familiar with marlin.

**Comment by Kalus1** [comment]
acceleration settings
axis    print      travel    
xy       1000    2000
z         100      100    
jerk xy  20.0  z 0.3
result:   skip on 1 layer

xy      500      1000
z        100       100
jerk xy 20.0  z 0.3
result:   good print

I think I will stick with the last settings and see how it goes, thank you for all of your help.

**Comment by errolt** [comment]
Just a thought. It might be noise on the Y limit switch. 

There is a setting somewhere to only check the limit switch on homing. Enable that.

**Comment by Kulitorum** [comment]
Yep, that was it for me, thanks.

**Comment by tiwanacote** [comment]
Hi!
I'm using a Rumba board and I have the same problem, layers missaligned only in the X direction.  At first it was in Y and X axis until  I changed the driver Y to the E2 posittion (always one axis at  time, courious) and now y X.

I setted ALWAYS_CHECK_ENDSTOPS to 0 so end-switchs are not the problem. .
Mechanically there is no problems.
Now I will change X driver to E1 position (what is the problem with rumba? There are some post to recommend?)

Any other idea?

Thanks!
